Embrace Unity iOS Interface/SpanRepository.swift

```diff
@@ -0,0 +1,92 @@
+//  Essentially identical to RN SpanRepository.Swift
+
+import Foundation
+import OSLog
+import OpenTelemetryApi
+
+// Should not get hit under normal circumstances, add as a guard against misinstrumentation
+private let MAX_STORED_SPANS = 10000
+
+class SpanRepository {
+    private let activeSpansQueue = DispatchQueue(label: "io.embrace.Unity.activeSpans", 
+                                                 attributes: .concurrent)
+    private let completedSpansQueue = DispatchQueue(label: "io.embrace.Unity.completedSpans", 
+                                                    attributes: .concurrent)
+    private var activeSpans = [String: Span]()
+    private var completedSpans = [String: Span]()
+    private var log = OSLog(subsystem: "Embrace", category: "SpanRepository")
+    
+    init() {
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(onSessionEnded),
+            name: Notification.Name("embrace.session.will_end"),
+            object: nil)
+    }
+    
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+    
+    func get(spanId: String) -> Span? {
+        var span: Span?
+        
+        activeSpansQueue.sync {
+            span = activeSpans[spanId]
+        }
+        
+        if span == nil {
+            completedSpansQueue.sync {
+                span = completedSpans[spanId]
+            }
+        }
+        
+        return span
+    }
+    
+    func spanStarted(span: Span) -> String {
+        var key = getKey(span: span)
+        
+        activeSpansQueue.async(flags: .barrier) {
+            if self.activeSpans.count > MAX_STORED_SPANS {
+                os_log("Embrace SpanRepository.swift --> Too many active spans being tracked. Ignoring new active span. Please consider reducing the number of spans being tracked.",
```

Comment:
I'm pretty sure `os_log` will grab the file where this log came from automatically. This prefix may not be necessary. No harm for duplication though